packages/lib/src/interactions/hooks.ts

```diff
@@ -5,46 +5,67 @@ import { useCallback, useEffect } from 'react';
 import { Vector2, Vector3 } from 'three';
 
 import { useVisCanvasContext } from '../vis/shared/VisCanvasProvider';
-import { getWorldFOV, htmlToWorld } from '../vis/utils';
+import { htmlToWorld } from '../vis/utils';
 import { useInteractionsContext } from './InteractionsProvider';
+import Box from './box';
 import type {
   CanvasEvent,
   CanvasEventCallbacks,
   InteractionEntry,
   ModifierKey,
 } from './models';
-import { clampPositionToArea } from './utils';
 
 const ZOOM_FACTOR = 0.95;
 const ONE_VECTOR = new Vector3(1, 1, 1);
 const MODIFIER_KEYS: ModifierKey[] = ['Alt', 'Control', 'Shift'];
 
 export function useMoveCameraTo() {
-  const { visSize } = useVisCanvasContext();
+  const { canvasSize, visSize } = useVisCanvasContext();
 
   const camera = useThree((state) => state.camera);
   const invalidate = useThree((state) => state.invalidate);
 
   return useCallback(
     (worldPt: Vector3) => {
-      const { position } = camera;
+      const { position, scale } = camera;
 
-      const { topRight, bottomLeft } = getWorldFOV(camera);
-      const width = Math.abs(topRight.x - bottomLeft.x);
-      const height = Math.abs(topRight.y - bottomLeft.y);
+      const fovWidth = canvasSize.width * scale.x;
+      const fovHeight = canvasSize.height * scale.y;
 
-      const clampedPosition = clampPositionToArea(
-        worldPt,
-        { width, height },
-        visSize
+      // Create box to clamp target camera position, to disallow panning outside of visualization bounds
+      const clampingBox = Box.empty(new Vector3(0, 0, position.z)).expandBySize(
+        Math.max(visSize.width - fovWidth, 0),
+        Math.max(visSize.height - fovHeight, 0)
```

Comment:
This is a similar computation as in `keepWithin`, but the center of the clamping box is different. Instead of trying to abstract things like we had done with `clampPositionToArea`, I kept the duplication. We can revisit this later as well if needed.

Reply:
Yeah, but with `keepWithin`, we'd be creating a box, clamping the center of that box to another box, subtracting the original center to the clamped point to get a `shift` vector, and then translating the box by that vector, only to get the center of the translated box...  Whereas in `moveCameraTo`, all we only need is to clamp a point to a box, that's it.

Basically, using the `Box` abstraction in `moveCameraTo` feels overkill and inefficient.

I could create a helper such as `getFovSize(camera, context)` to abstract away the `canvasSize * scale` computation, at least... And maybe let `expandBySize` accept a `Size` object instead of two values... Not sure.

Reply:
> Basically, using the Box abstraction in moveCameraTo feels overkill and inefficient.

In terms of implementation perhaps but conceptually, I think it makes a lot of sense. You are already creating a `clampingBox` so there is only the need for a a second one:

```ts
      // Similar to `clampingBox`
      const visBox = new Box().setFromCenterAndSize(
        new Vector3(0, 0, position.z),
        new Vector3(visSize.width, visSize.height, 0)
      );
      // New box to be added
      const fovBox = new Box().setFromCenterAndSize(
        worldPt,
        new Vector3(fovWidth, fovHeight, 0)
      );
     
      // Actual operation
      fovBox.keepWithin(visBox);

      position.copy(zoomBox.center);
```

Having a `setFromCenterAndSize` that accepts a `Size` object rather than a `Vector3` could also help simplifying the code.

Reply:
Yeah, alright, though I prefer the API `Box.empty(position).expandBySize(width, height)`. I could add a static method like `Box.fromCenterAndSize()`, but it seems like it should accept two arguments, not three, as I like passing the size as two values.

Reply:
Done, much nicer indeed. I've added a static method called `Box.fromSize(size: Size)` to create a box with a given size and centered on `(0, 0, 0)`. If the need is for a box with a different center, then `Box.empty(center).expandBySize(width, height)` is still the way to go.

Reply:
Fine for me. You merge whenever you want :octopus: